src/main/java/at/hannibal2/skyhanni/data/MaxwellAPI.kt

```diff
@@ -0,0 +1,124 @@
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.events.RepositoryReloadEvent
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.NumberUtil.formatNumber
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.StringUtils.removeColor
+import at.hannibal2.skyhanni.utils.StringUtils.removeResets
+import at.hannibal2.skyhanni.utils.StringUtils.trimWhiteSpace
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+object MaxwellAPI {
+    private val storage get() = ProfileStorageData.profileSpecific
+    var currentPower: String?
+        get() = storage?.maxwell?.currentPower
+        set(value) {
+            storage?.maxwell?.currentPower = value ?: return
+        }
+    var magicalPower: Int?
+        get() = storage?.maxwell?.magicalPower
+        set(value) {
+            storage?.maxwell?.magicalPower = value ?: return
+        }
+
+    private var powers = mutableListOf<String>()
+
+    private val group = RepoPattern.group("data.maxwell")
+    private val chatPowerpattern by group.pattern(
+        "chat.power",
+        "§eYou selected the §a(?<power>.*) §e(power )?for your §aAccessory Bag§e!"
+    )
+    private val inventoryPowerPattern by group.pattern(
+        "inventory.power",
+        "§7Selected Power: §a(?<power>.*)"
+    )
+    private val inventoryMPPattern by group.pattern(
+        "inventory.magicalpower",
+        "§7Magical Power: §6(?<mp>[\\d,]+)"
+    )
+    private val thaumaturgyGuiPattern by group.pattern(
+        "gui.thaumaturgy",
+        "Accessory Bag Thaumaturgy"
+    )
+    private val yourBagsGuiPattern by group.pattern(
+        "gui.yourbags",
+        "Your Bags"
+    )
+    private val powerSelectedPattern by group.pattern(
+        "gui.selectedpower",
+        "§aPower is selected!"
+    )
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!isEnabled()) return
+        val message = event.message.trimWhiteSpace().removeResets()
+
+        chatPowerpattern.matchMatcher(message) {
+            val power = group("power")
+            currentPower = getPowerByNameOrNull(power) ?: return
+            return
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryFullyLoaded(event: InventoryFullyOpenedEvent) {
+        if (!isEnabled()) return
+
+        if (thaumaturgyGuiPattern.matches(event.inventoryName)) {
+            val stacks = event.inventoryItems
+            val selectedPowerStack =
+                stacks.values.find {
+                    val lore = it.getLore()
+                    lore.isNotEmpty() && powerSelectedPattern.matches(lore.last())
```

Comment:
Does the same but more compact, since parsing a null to Pattern.matches returns false
```suggestion
                    powerSelectedPattern.matches(lore.lastOrNull())
```